From linux-usb-devel-bounces@lists.sourceforge.net Wed Dec 13 00:04:01 2006
Date: Wed, 13 Dec 2006 00:03:38 -0800
From: Dan Carpenter <error27@gmail.com>
To: linux-usb-devel@lists.sourceforge.net
Message-ID: <20061213080338.GA19096@localhost.localdomain>
MIME-Version: 1.0
Content-Disposition: inline
Cc: zaitcev@redhat.com
Subject: USB: devio.c add missing INIT_LIST_HEAD()
Content-Type: text/plain; charset="us-ascii"

From: Dan Carpenter <error27@gmail.com>

It should hopefully fix the list corruption bug on:
https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=214402

Add a missing INIT_LIST_HEAD()

Signed-off-by: Dan Carpenter <error27@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/devio.c |    1 +
 1 file changed, 1 insertion(+)

--- gregkh-2.6.orig/drivers/usb/core/devio.c
+++ gregkh-2.6/drivers/usb/core/devio.c
@@ -570,6 +570,7 @@ static int usbdev_open(struct inode *ino
 	ps->dev = dev;
 	ps->file = file;
 	spin_lock_init(&ps->lock);
+	INIT_LIST_HEAD(&ps->list);
 	INIT_LIST_HEAD(&ps->async_pending);
 	INIT_LIST_HEAD(&ps->async_completed);
 	init_waitqueue_head(&ps->wait);
